discovery/docker_discovery.go

```diff
@@ -163,7 +181,49 @@ func (d *DockerDiscovery) getContainers() {
 			continue
 		}
 
-		svc := service.ToService(&container)
+		n := &namer{
```

Comment:
Instead of doing it this way, I'd make an interface called "Namer" like:
```
type Namer interface {
    Name(*service.Service) string
}
```
You can then make two implementations like:
```
type RegexNamer struct {}

func (r *RegexNamer) Name(svc *service.Service) string {
    doSomethingHere()
}

type LabelNamer struct {}

func (l *LabelNamer) Name(svc *service.Service) { 
    blahBlah()
}
```

That will be a lot cleaner and easier to pass around.